api/supabase/functions/timeline/index.ts

```diff
@@ -0,0 +1,45 @@
+import "https://esm.sh/@supabase/functions-js/src/edge-runtime.d.ts"
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
+
+Deno.serve(async (req) => {
+  const supaURL: string | undefined = Deno.env.get("SUPABASE_URL")
+  const supaAnonKey: string | undefined = Deno.env.get("SUPABASE_ANON_KEY")
+
+  let supabase: any
+  if (supaURL !== undefined && supaAnonKey !== undefined) {
+    supabase = createClient(supaURL, supaAnonKey)
+  }
+
+  // user情報を取得
+  let token = req.headers.get('authorization')
+  if (token != null) {
+    token = token.split('Bearer ')[1];
+  }
+  const userInfo = await supabase.auth.getUser(token);
```

Comment:
[imo]
こちらも`null`の場合は例外扱いにしてしまったほうがシンプルになるかなと思います。(認証通っていない時点で音源情報取得する必要が無いと思うので)
```suggestion
  if (token === null) {
    throw new Error('unauthorized');
  }
  const userInfo = await supabase.auth.getUser(token.split('Bearer ')[1];);
```

Reply:
その通りだと思いますので、いただいたコードで修正しました。